subsys/net/l2/ethernet/gptp/gptp_mi.c

```diff
@@ -734,16 +734,22 @@ static void gptp_update_local_port_clock(void)
 {
 	struct gptp_clk_slave_sync_state *state;
 	struct gptp_global_ds *global_ds;
+	struct gptp_servo_ds *servo_ds;
 	struct gptp_port_ds *port_ds;
 	int port;
-	int64_t nanosecond_diff;
-	int64_t second_diff;
+	uint64_t sync_receipt_ts;
```

Comment:
I suggest you use net_time_t for ns precision timestamps where appropriate, see the definition in net_time.h and ptp_time.h. (Think Linux's "ktime_t" whenever you read net_time_t.)